File: backend/src/main/java/com/horain/llm/PlaceholderLlmClient.java
package com.horain.llm;

import java.util.List;

/**
 * Placeholder LLM client when no real provider is configured.
 * Returns canned responses for testing without an API key.
 */
public class PlaceholderLlmClient implements LlmClient {

    @Override
    public boolean isConfigured() {
        return false;
    }

    @Override
    public LlmResponse chat(List<ChatMessage> messages, List<ToolDefinition> tools) {
        String lastUserMessage = "";
        for (int i = messages.size() - 1; i >= 0; i--) {
            ChatMessage m = messages.get(i);
            if ("user".equals(m.role()) && m.content() != null && !m.content().isBlank()) {
                lastUserMessage = m.content();
                break;
            }
        }

        return new LlmResponse(
                "LLM is not configured. Set LLM_API_KEY (and optionally LLM_BASE_URL, LLM_MODEL) to enable the assistant. " +
                "Your message was: \"" + lastUserMessage + "\"",
                null,
                "stop"
        );
    }
}

File: backend/src/main/kotlin/com/horain/llm/PlaceholderLlmClient.kt
package com.horain.llm

/**
 * Placeholder LLM client when no real provider is configured.
 * Returns canned responses for testing without an API key.
 */
class PlaceholderLlmClient : LlmClient {

    override fun isConfigured(): Boolean = false

    override fun chat(messages: List<ChatMessage>, tools: List<ToolDefinition>): LlmResponse {
        var lastUserMessage = ""
        for (i in messages.indices.reversed()) {
            val m = messages[i]
            if (m.role.equals("user", ignoreCase = true) && !m.content.isNullOrBlank()) {
                lastUserMessage = m.content!!
                break
            }
        }
        return LlmResponse(
            "LLM is not configured. Set LLM_API_KEY (and optionally LLM_BASE_URL, LLM_MODEL) to enable the assistant. " +
                "Your message was: \"$lastUserMessage\"",
            null,
            "stop"
        )
    }
}
